src/transport/include/antioch/Stockmayer_potential.h

```diff
@@ -0,0 +1,239 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+//--------------------------------------------------------------------------
+
+#ifndef ANTIOCH_STOCKMAYER_POTENTIAL_H
+#define ANTIOCH_STOCKMAYER_POTENTIAL_H
+
+//Antioch
+#include "antioch/antioch_asserts.h"
+#include "antioch/metaprogramming_decl.h"
+#include "antioch/cmath_shims.h"
+
+//C++
+
+namespace Antioch{
+
+
+  /* Stockmayer potential from
+     Monchich & Mason 1961, surface[T][delta]
+   */
+  template <typename CoeffType> //typename VectorCoeffType, typename MatrixCoeffType>
+  class StockmayerPotential
+  {
+        public:
+          StockmayerPotential();
+          ~StockmayerPotential();
+
+          const std::vector<CoeffType> temperature()     const {return _T;}
+
+          const std::vector<CoeffType> log_temperature() const {return _logT;}
+
+          const std::vector<CoeffType> delta()           const {return _delta;}
+
+          const std::vector<std::vector<CoeffType> > omega_1_1() const {return _omega_1_1;}
+          const std::vector<std::vector<CoeffType> > omega_2_2() const {return _omega_2_2;}
+
+        private:
+
+          void init();
+
+
+          template <typename Scalar>
+          const Scalar omega_2_2(const Scalar & T_red, const Scalar & dipole_red) const {antioch_not_implemented();}
+
+          template <typename Scalar>
+          const Scalar omega_1_1(const Scalar & T_red, const Scalar & dipole_red) const {antioch_not_implemented();}
+
+          const unsigned int _T_size;
+          const unsigned int _delta_size;
+
+          std::vector<CoeffType>               _T, _logT;
+          std::vector<CoeffType>               _delta;
+          std::vector<std::vector<CoeffType> > _omega_1_1;
+          std::vector<std::vector<CoeffType> > _omega_2_2;
+
+  };
+
+
+  template <typename CoeffType>
+  inline
+  StockmayerPotential<CoeffType>::StockmayerPotential():
+        _T_size(37),
+        _delta_size(8),
+        _T(_T_size,0.),_logT(_T_size,0.),
+        _delta(_delta_size,0.),
+        _omega_1_1(_T_size,std::vector<CoeffType>(_delta_size,0.)),
+        _omega_2_2(_T_size,std::vector<CoeffType>(_delta_size,0.))
+
+  {
+     this->init();
+
+     return;
+  }
+
+  template <typename CoeffType>
+  inline
+  StockmayerPotential<CoeffType>::~StockmayerPotential()
+  {
+     return;
+  }
+
+  template <typename CoeffType>
+  inline
+  void StockmayerPotential<CoeffType>::init()
```

Comment:
So StockmayerPotential is not just a function, but it is also number? I.e. the flood of hardcoded data is always tied to this and  StockmayerPotential will never mean anything else?